packages/eslint-plugin-svelte/src/utils/svelte-context.ts

```diff
@@ -0,0 +1,201 @@
+import type { RuleContext } from '../types.js';
+import fs from 'fs';
+import path from 'path';
+import { getPackageJson } from './get-package-json.js';
+import { getFilename, getSourceCode } from './compat.js';
+
+const isRunInBrowser = !fs.readFileSync;
+
+export type SvelteContext = {
+	svelteVersion: '3/4' | 5;
+	fileType: '.svelte' | '.svelte.[js|ts]';
+	runes: boolean;
+	svelteKitVersion: '1-next' | 1 | 2 | null;
+	svelteKitFileType:
+		| '+page.svelte'
+		| '+page.js'
+		| '+page.server.js'
+		| '+error.svelte'
+		| '+layout.svelte'
+		| '+layout.js'
+		| '+layout.server.js'
+		| '+server.js'
+		| null;
+};
+
+function getFileType(filePath: string): SvelteContext['fileType'] | null {
+	if (filePath.endsWith('.svelte')) {
+		return '.svelte';
+	}
+
+	if (filePath.endsWith('.svelte.js') || filePath.endsWith('.svelte.ts')) {
+		return '.svelte.[js|ts]';
+	}
+
+	return null;
+}
+
+function getSvelteKitFileTypeFromFilePath(filePath: string): SvelteContext['svelteKitFileType'] {
+	const fileName = filePath.split('/').pop();
+	switch (fileName) {
+		case '+page.svelte': {
+			return '+page.svelte';
+		}
+		case '+page.js':
+		case '+page.ts': {
+			return '+page.js';
+		}
+		case '+page.server.js':
+		case '+page.server.ts': {
+			return '+page.server.js';
+		}
+		case '+error.svelte': {
+			return '+error.svelte';
+		}
+		case '+layout.svelte': {
+			return '+layout.svelte';
+		}
+		case '+layout.js':
+		case '+layout.ts': {
+			return '+layout.js';
+		}
+		case '+layout.server.js':
+		case '+layout.server.ts': {
+			return '+layout.server.js';
+		}
+		case '+server.js':
+		case '+server.ts': {
+			return '+server.js';
+		}
+		default: {
+			return null;
+		}
+	}
+}
+
+function getSvelteKitContext(
+	context: RuleContext
+): Pick<SvelteContext, 'svelteKitFileType' | 'svelteKitVersion'> {
+	const filePath = getFilename(context);
+	const svelteKitVersion = getSvelteKitVersion(filePath);
+	if (svelteKitVersion == null) {
+		return {
+			svelteKitFileType: null,
+			svelteKitVersion: null
+		};
+	}
+	if (isRunInBrowser) {
```

Comment:
Sorry to duplicate the comment, the other one was not part of the same review :/

Is the plugin still run in the browser? I thought that functionality was removed. If this is really needed, I think this is not the place to do it - instead the whole `SvelteContext` should be replaced by a mock version when running in the browser playground. That way it is clearly separated from the logic reading things like sveltekit versions...

Reply:
Hum, Even so, I personally still want to support [vite-plugin-eslint4b](https://github.com/ota-meshi/vite-plugin-eslint4b)

Reply:
Yeah, that makes sense. But then I'd mock the whole `SvelteContext` somewhere else (probably in `getSvelteContext`) and keep this function simple

Reply:
It might just be personal preference, but I see the possibility of this program running in a browser as part of its "spec." That’s why I implemented the `getSvelteKitContext` function to support behavior in the browser. In any case, it’s probably just a matter of preference.

Reply:
> It might just be personal preference, but I see the possibility of this program running in a browser as part of its "spec."

Yeah, I agree. But I don't think this is the right place to implement the check. I think either `shouldRun` should always return true in the browser, or maybe the context could be configurable in the browser environment...

Reply:
I don't think this is the scope of this PR. Also, since the code for browsers is not much complex and not a major use case at this time, it should be modified when this need is further increased or the complexity rises. If you have suggestions for improvement, we would welcome another PR.